flaml/automl/automl.py

```diff
@@ -3550,6 +3659,10 @@ def _search(self):
                         and ray.available_resources()["CPU"]
                         or os.cpu_count()
                     )
+                elif self._use_spark is not False:
+                    from flaml.utils import get_n_cpus
+
+                    n_cpus = get_n_cpus()
```

Comment:
Do you have a test that covers this line? That is, both use_spark and ensemble=True.
Also, here the ensemble is run in the driver node. But `get_n_cpus()` returns the number of cpus in worker nodes.
Consider doing the ensemble in worker nodes in future?

Reply:
Test added, get_n_cpus() updated.